protobuf/lib/src/protobuf/extension_field_set.dart

```diff
@@ -177,7 +177,7 @@ class _ExtensionFieldSet {
     _isReadOnly = true;
     for (var field in _info.values) {
       if (field.isRepeated) {
-        final entries = _values[field.tagNumber];
+        PbList? entries = _values[field.tagNumber];
```

Comment:
Should we move this type cast after `entries == null` check below? Assuming null checks are the same for every type, this would avoid casting in AOT and JIT when the value is `null`.

Reply:
Good idea.

Yes, a `as PbList?` is more expensive than ` == null`.

Reply:
My understanding from our discussion on this:

- When we cast, then check for null:
  1. Check for cast: `valueClassId == NullClassId || valueClassId == PbListClassId`
  2. Then check if the value is null
 
- When we check for null, then cast:
  1. Check if the value is null (same check as the one above)
  2. Check for cast: `valueClassId == PbListClassId`

It shouldn't make a huge difference, but it seems like checking for null before casting should need a few instructions less.